packages/bsp-app-download-sub-generator/src/app/index.ts

```diff
@@ -0,0 +1,351 @@
+            this.log,
+            this.options.logWrapper,
+            this.options.logLevel,
+            this.options.logger,
+            this.vscode
+        );
+
+        //this.prompts = new Prompts([]);
+        // Initialise prompts and callbacks if not launched as a subgenerator
+        this.appWizard.setHeaderTitle(generatorTitle);
+        this.prompts = new Prompts(getYUIDetails());
+        this.setPromptsCallback = (fn): void => {
+            if (this.prompts) {
+                this.prompts.setCallback(fn);
+            }
+        };
+    }
+
+    /**
+     * Initialises necessary settings and telemetry for the generator.
+     */
+    public async initializing(): Promise<void> {
+        if ((this.env as unknown as YeomanEnvironment).conflicter) {
+            (this.env as unknown as YeomanEnvironment).conflicter.force = this.options.force ?? true;
+        }
+
+        // Initialise telemetry settings
+        await TelemetryHelper.initTelemetrySettings({
+            consumerModule: {
+                name: generatorName,
+                version: this.rootGeneratorVersion()
+            },
+            internalFeature: isInternalFeaturesSettingEnabled(),
+            watchTelemetrySettingStore: false
+        });
+    }
+
+    /**
+     * Prompts the user for application details and downloads the app.
+     */
+    public async prompting(): Promise<void> {
+        const quickDeployedAppConfig = this.options?.data?.quickDeployedAppConfig;
+        const questions: BspAppDownloadQuestions[] = await getPrompts(this.appRootPath, quickDeployedAppConfig);
+        const answers: BspAppDownloadAnswers = await this.prompt(questions);
+        const { targetFolder } = answers;
+        if (quickDeployedAppConfig?.appId) {
+            // Handle quick deployed app download where prompts for system selection and app selection are not displayed
+            // Only target folder prompt is shown
+            await this._handleQuickDeployedAppDownload(quickDeployedAppConfig, targetFolder);
+        } else {
+            // Handle app download where prompts for system selection and app selection are shown
+            Object.assign(this.answers, answers);
+        }
+    }
+
+    /**
+     *
+     * @param quickDeployedAppConfig - The configuration for the quick deployed app.
+     * @param targetFolder - The target folder where the app will be downloaded.
+     */
+    private async _handleQuickDeployedAppDownload(
+        quickDeployedAppConfig: QuickDeployedAppConfig,
+        targetFolder: string
+    ): Promise<void> {
+        const appList = await fetchAppListForSelectedSystem(
+            quickDeployedAppConfig.serviceProvider,
+            quickDeployedAppConfig.appId
+        );
+        if (!appList.length) {
+            BspAppDownloadLogger.logger?.error(
+                t('error.quickDeployedAppDownloadErrors.noAppsFound', { appId: quickDeployedAppConfig.appId })
+            );
+            throw new Error();
```

Comment:
I just wonder in practice, if the user has gone through the step of selecting the location for their new app to be saved, at that point I think the assumption would be that the details have been fetched, but it would throw an error after they click `Finish` 

If POs are ok with this, I think the error needs to be more descriptive